### PR TITLE
Prefill billable

### PR DESCRIPTION
**Pull request by pradeep8093**

### CODE REVIEW

File: src/app/core/services/expense-fields.service.ts [hunk 1]
```diff
@@ -29,6 +29,19 @@ export class ExpenseFieldsService {
     );
   }
 
+  formatBillableFields(expenseFields: Array<ExpenseField>) {
```
Comment (Julias0) [inline src/app/core/services/expense-fields.service.ts:32]: ExpenseField[]

File: src/app/core/services/expense-fields.service.ts [hunk 2]
```diff
@@ -29,6 +29,19 @@ export class ExpenseFieldsService {
...
+    return expenseFields.map(function (field) {
+      if (!field.is_custom && field.field_name.toLowerCase() === 'billable') {
+        if (field.default_value === 'true') {
```
Comment (Julias0) [inline src/app/core/services/expense-fields.service.ts:35]: field.default_value = field.default_value === 'true'

File: src/app/core/services/expense-fields.service.ts [hunk 3]
```diff
@@ -47,6 +60,8 @@ export class ExpenseFieldsService {
         expenseFields => {
           const expenseFieldMap: Partial<ExpenseFieldsMap> = {};
 
+          this.formatBillableFields(expenseFields);
```
Comment (Julias0) [inline src/app/core/services/expense-fields.service.ts:63]: expenseFields = this.formatBillableFields(expenseFields);

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -935,7 +935,8 @@ export class AddEditExpensePage implements OnInit {
       instaFyleSettings: instaFyleSettings$,
       imageData: this.getInstaFyleImageData(),
       recentCurrency: from(this.recentLocalStorageItemsService.get('recent-currency-cache')),
-      recentValue: this.recentlyUsedValues$
+      recentValue: this.recentlyUsedValues$,
+      txnFields: this.txnFields$
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:939]: This creates a cyclic loop. Lets do this elsewhere

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 5]
```diff
@@ -957,7 +957,6 @@ export class AddEditExpensePage implements OnInit {
         if (!bankTxn) {
           etxn = {
             tx: {
-              billable: false,
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:960]: why remove this?

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 6]
```diff
@@ -721,7 +726,6 @@ export class AddEditMileagePage implements OnInit {
         }
         return {
           tx: {
-            billable: false,
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:724]: same as above

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 5 above)
Comment (pradeep8093) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:960]: so after setting default it was getting reset here to false

**Review by Julias0 — APPROVED** [review]

**Pull request merged**